### PR TITLE
[stable/prometheus] Add custom pod labels for prom server

### PR DESCRIPTION
**Pull request by OGKevin**
#### What this PR does / why we need it:
This PR makes it possible to add custom, user defined, labels to the prometheus server component.
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by OGKevin** [comment]
@mgaggero @gianrubio 👋 any update regarding this?

**Comment by mgaggero** [comment]
> @mgaggero @gianrubio 👋 any update regarding this?

I guess there is a typo in one of the names.

**Comment by OGKevin** [comment]
> > @mgaggero @gianrubio 👋 any update regarding this?
> 
> I guess there is a typo in one of the names.

ah there is. my bad. I meant to tag @mgoodness

**Comment by OGKevin** [comment]
@maorfr Could you please explain to me why this is failing? The details in the provided link don't tell me anything. What I can tell is that the helm deployment timed out? If so could you see why?

File: stable/prometheus/templates/server-statefulset.yaml [hunk 1]
```diff
@@ -9,6 +9,7 @@ metadata:
   labels:
     {{- include "prometheus.server.labels" . | nindent 4 }}
   name: {{ template "prometheus.server.fullname" . }}
+    {{ toYaml .Values.server.statefulSet.labels | indent 4 }}
```
Comment (maorfr) [inline stable/prometheus/templates/server-statefulset.yaml]: i think that this should move 1 line up

Comment (OGKevin) [reply stable/prometheus/templates/server-statefulset.yaml]: Hmm if I remember correctly it worked for me let me double check then.

Comment (OGKevin) [reply stable/prometheus/templates/server-statefulset.yaml]: Found it, I did not test if they were empty 🤦‍♂ my bad.

**Comment by maorfr** [comment]
/lgtm

nicely done, thanks!

**Pull request merged**